### PR TITLE
Improve ansible-galaxy error for InconsistentCandidate exception

### LINKED ISSUE #75139: Installing collection artifact with mismatch version results in exception
### Summary

When installing a collection artifact from a URL or tarball, with a mismatched `version`, an exception is raised, that should be handled and a better message provided to the user.

### Issue Type

Bug Report

### Component Name
```
lib/ansible/galaxy/collection/__init__.py
```
### Ansible Version

```console
$ ansible --version
2.11+
```

### Configuration

```console
$ ansible-config dump --only-changed
```

### OS / Environment

N/A

### Steps to Reproduce

```shell
ansible-galaxy collection download -p . --no-deps sivel.toiletwater:==0.0.10
sed -i 's/version: 0.0.10/version: 0.0.9/' requirements.yml
ansible-galaxy collection install -p . -r requirements.yml -vvv
```

### Expected Results

Friendly error message

### Actual Results

```console
ERROR! Unexpected Exception, this is probably a bug: Provided candidate <sivel.toiletwater:0.0.10 of type 'file' from sivel-toiletwater-0.0.10.tar.gz> does not satisfy <sivel.toiletwater:0.0.9 of type 'file' from sivel-toiletwater-0.0.10.tar.gz>
the full traceback was:

Traceback (most recent call last):
  File "/Users/matt/projects/ansibledev/ansible/bin/ansible-galaxy", line 133, in <module>
    exit_code = cli.run()
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/cli/galaxy.py", line 552, in run
    return context.CLIARGS['func']()
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/cli/galaxy.py", line 75, in method_wrapper
    return wrapped_method(*args, **kwargs)
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/cli/galaxy.py", line 1186, in execute_install
    self._execute_install_collection(
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/cli/galaxy.py", line 1213, in _execute_install_collection
    install_collections(
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/galaxy/collection/__init__.py", line 512, in install_collections
    dependency_map = _resolve_depenency_map(
  File "/Users/matt/projects/ansibledev/ansible/lib/ansible/galaxy/collection/__init__.py", line 1329, in _resolve_depenency_map
    return collection_dep_resolver.resolve(
  File "/Users/matt/venvs/ansibledev/lib/python3.9/site-packages/resolvelib/resolvers.py", line 453, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "/Users/matt/venvs/ansibledev/lib/python3.9/site-packages/resolvelib/resolvers.py", line 347, in resolve
    failure_causes = self._attempt_to_pin_criterion(name, criterion)
  File "/Users/matt/venvs/ansibledev/lib/python3.9/site-packages/resolvelib/resolvers.py", line 221, in _attempt_to_pin_criterion
    raise InconsistentCandidate(candidate, criterion)
resolvelib.resolvers.InconsistentCandidate: Provided candidate <sivel.toiletwater:0.0.10 of type 'file' from sivel-toiletwater-0.0.10.tar.gz> does not satisfy <sivel.toiletwater:0.0.9 of type 'file' from sivel-toiletwater-0.0.10.tar.gz>
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Fixes #75139

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

File: changelogs/fragments/75235-ansible-galaxy-inconsistent-candidate-error.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - ansible-galaxy - Improve error message from dependency resolution when a candidate has inconsistent requirements.
```
Comment (Akasurde) [inline changelogs/fragments/75235-ansible-galaxy-inconsistent-candidate-error.yml]:
```suggestion
  - ansible-galaxy - Improve error message from dependency resolution when a candidate has inconsistent requirements (https://github.com/ansible/ansible/issues/75139).
```

**Review by sivel — APPROVED** [review]

File: lib/ansible/galaxy/collection/__init__.py [hunk 2]
```diff
@@ -1353,3 +1354,29 @@ def _resolve_depenency_map(
...
+        error_msg_lines = [
+            (
+                'Failed to resolve the requested dependencies map. '
+                'Got the candidate {req.fqcn!s}:{req.ver!s} ({dep_origin!s}) '
+                'which didn\'t satisfy all of the following requirements:'.
+                format(
+                    req=dep_exc.candidate,
+                    dep_origin='direct request'
+                    if not parents else 'dependency of {parent!s}'.
+                    format(parent=','.join(parents))
```
Comment (webknjaz) [inline lib/ansible/galaxy/collection/__init__.py]: How about having a whitespace after the commas?
```suggestion
                    if not parents else 'dependency of {parents!s}'.
                    format(parents=', '.join(parents))
```

File: test/integration/targets/ansible-galaxy-collection/tasks/install.yml [hunk 3]
```diff
@@ -163,6 +163,36 @@
...
+
+- assert:
+    that: error == expected_error
+  vars:
+    reset_color: '\x1b\[0m'
+    color: '\x1b\[[0-9];[0-9]{2}m'
+    error: "{{ result.stderr | regex_replace(reset_color) | regex_replace(color) | regex_replace('\\n', ' ') }}"
+    expected_error: >-
+            ERROR! Failed to resolve the requested dependencies map.
+            Got the candidate namespace3.name:1.0.0 (direct request)
```
Comment (webknjaz) [inline test/integration/targets/ansible-galaxy-collection/tasks/install.yml:192]: It'd be useful to have a "dependency of" test case too.

Comment (s-hertel) [reply test/integration/targets/ansible-galaxy-collection/tasks/install.yml:192]: This caught a bug, thanks!

**Comment by s-hertel** [comment]
Thanks for the reviews!

**Pull request merged**